cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/EntityOperatorReconciler.java

```diff
@@ -117,13 +122,47 @@ public Future<Void> reconcile(boolean isOpenShift, ImagePullPolicy imagePullPoli
                 .compose(i -> userOperatorRoleBindings())
                 .compose(i -> topicOperatorConfigMap())
                 .compose(i -> userOperatorConfigMap())
+                .compose(i -> topicOperatorCruiseControlApiSecret())
                 .compose(i -> deleteOldEntityOperatorSecret())
                 .compose(i -> topicOperatorSecret(clock))
                 .compose(i -> userOperatorSecret(clock))
                 .compose(i -> deployment(isOpenShift, imagePullPolicy, imagePullSecrets))
                 .compose(i -> waitForDeploymentReadiness());
     }
 
+    /**
+     * Manages the Cruise Control API keys secret used by the Topic Operator.
+     *
+     * @return Future which completes when the reconciliation is done.
+     */
+    protected Future<Void> topicOperatorCruiseControlApiSecret() {
+        if (cruiseControlEnabled) {
+            String ccApiSecretName = KafkaResources.entityTopicOperatorCcApiSecretName(reconciliation.name());
+            if (entityOperator != null && entityOperator.topicOperator() != null) {
+                return secretOperator.getAsync(reconciliation.namespace(), ccApiSecretName)
+                    .compose(oldSecret -> {
+                        Secret newSecret = entityOperator.topicOperator().generateCruiseControlApiSecret();
+
+                        if (oldSecret != null) {
+                            // The credentials should not change with every release
+                            // So if the secret with credentials already exists, we re-use the values
+                            // But we use the new secret to update labels etc. if needed
+                            newSecret.setData(oldSecret.getData());
+                        }
```

Comment:
Again, this seems to be very inefficient. You can pass the old password from the old secret into the Model class secret generator.

Reply:
Done.